### DISCUSSION TITLE
Minimizing LFS impact on a history rewrite

### DISCUSSION THREAD

**Discussion by richard-fine** [git-lfs/git-lfs#5450] [Q&A] [answered]
I'm carrying out an extensive history-rewriting operation on a repository which uses LFS files (converting regular merges into squash-merges, changing the commit messages, etc). This necessarily involves checking out each individual revision in the process, and I would like to avoid spending any time downloading and checking out LFS files as part of that.

I think that in principle it shouldn't be necessary to pull the LFS files, and that I should just be able to cherry-pick through the pointer files directly - I'm not changing any commit content or adding/removing LFS files, though I _am_ cherry-picking with `--no-commit` and then running `commit` separately (to control commit metadata).

What's the safe and most performant way to do this? Right now I am running `git lfs install --skip-smudge` on the repo before I start, which avoids pulling/checking out the files, but it looks like it still runs the `lfs clean` step at times and I also am not totally clear on the safety of it. Should I just be doing `git lfs uninstall` completely? Or is there a better approach I can take?

**Comment by chrisd8088** [comment]
I think your suggestion to consider temporarily uninstalling Git LFS might be a good one, or what I do sometimes for testing purposes is just comment out the lines in the `[filter "lfs"]` section of my`~/.gitconfig` file.  If you've got Git LFS hooks installed somewhere, it can be useful to hide those as well (again, I tend to just comment out their contents out if I need to simulate a system with Git LFS uninstalled, or move them to a temp directory, or something like that.).

**Reply by richard-fine** [reply]
Alright, cool - uninstalling it certainly seems like the most high-performance option :) 

Subsequently pushing the rewritten branch back to the repo on my GHES instance should also be totally safe, even without LFS installed, because the pointer files in my new branch will have identical content to the original branch, and so all the LFS files are already guaranteed to be on the server from the original branch. Correct?

**Reply by richard-fine** [reply]
Hm, so I tried `git lfs uninstall --local` to remove it from the repo, but I get failures during `cherry-pick` because it still tries to run the smudge filter. I assume it's because it's still installed globally; is there no way to have it installed globally but uninstalled for a specific repo?

**Reply by chrisd8088** [reply, marked as answer]
> Subsequently pushing the rewritten branch back to the repo on my GHES instance should also be totally safe, even without LFS installed, because the pointer files in my new branch will have identical content to the original branch, and so all the LFS files are already guaranteed to be on the server from the original branch. Correct?

I think that should be the case, yes.

> is there no way to have it installed globally but uninstalled for a specific repo?

Maybe try something like this in your repo's `.git/config`:
```
[filter "lfs"]
	process = ""
	required = false
	clean = ""
	smudge = ""
```
